src/txn/txn_recover.c

```diff
@@ -449,6 +452,9 @@ __recovery_setup_file(WT_RECOVERY *r, const char *uri, const char *config)
       (WT_IS_MAX_LSN(&r->max_ckpt_lsn) || __wt_log_cmp(&lsn, &r->max_ckpt_lsn) > 0))
         r->max_ckpt_lsn = lsn;
 
+    /* Update the base write gen based on this file's configuration. */
+    WT_RET(__wt_metadata_update_base_write_gen(r->session, config));
```

Comment:
@sueloverso, @quchenhao I was trying to convince myself we always call `__recovery_file_scan()`, and it turns out we call it in two different places in `__wt_txn_recover()`, and I couldn't figure out why. Couldn't we just call it from one place, before this line of code?
```
    /*
     * If no log was found (including if logging is disabled), or if the last checkpoint was done
     * with logging disabled, recovery should not run. Scan the metadata to figure out the largest
     * file ID.
     */
    if (!FLD_ISSET(conn->log_flags, WT_CONN_LOG_EXISTED) || WT_IS_MAX_LSN(&metafile->ckpt_lsn)) {
```

Reply:
That doesn't work for the backup code path. In backup, we need to first recover the metadata then call the function. Tried with your suggestion and some backup tests failed.